File: scratch/extract_docx.mjs
import fs from 'fs';
import { execSync } from 'child_process';

const docxPath = String.raw`C:\Users\Nishtha Yadav\Desktop\CANCELLATION POLICY.docx`;
const outPath = new URL('./cancellation_policy.txt', import.meta.url);

if (!fs.existsSync(docxPath)) {
  fs.writeFileSync(outPath, 'NOT FOUND: ' + docxPath);
  process.exit(1);
}

const script = `
import zipfile, xml.etree.ElementTree as ET
with zipfile.ZipFile(r"${docxPath.replace(/\\/g, '\\\\')}") as z:
    root = ET.fromstring(z.read('word/document.xml'))
ns = {'w': 'http://schemas.openxmlformats.org/wordprocessingml/2006/main'}
paras = []
for p in root.findall('.//w:p', ns):
    texts = [t.text for t in p.findall('.//w:t', ns) if t.text]
    if texts:
        paras.append(''.join(texts))
print('\\n'.join(paras))
`;

try {
  const text = execSync(`python -c "${script.replace(/"/g, '\\"').replace(/\n/g, '; ')}"`, { encoding: 'utf8' });
  fs.writeFileSync(outPath, text);
  console.log(text);
} catch (e) {
  fs.writeFileSync(outPath, 'ERROR: ' + (e.stderr || e.message));
  process.exit(1);
}

File: scratch/extract_docx.py
import zipfile
import xml.etree.ElementTree as ET
from pathlib import Path

src = Path(r"C:\Users\Nishtha Yadav\Desktop\CANCELLATION POLICY.docx")
dst = Path(r"C:\Users\Nishtha Yadav\Desktop\watch\scratch\cancellation_policy.txt")
dst.parent.mkdir(parents=True, exist_ok=True)

if not src.is_file():
    print("FILE_NOT_FOUND")
    raise SystemExit(1)

W = "{http://schemas.openxmlformats.org/wordprocessingml/2006/main}"
parts = []
with zipfile.ZipFile(src) as z:
    xml = z.read("word/document.xml")
root = ET.fromstring(xml)
for p in root.iter(W + "p"):
    texts = []
    for node in p.iter(W + "t"):
        if node.text:
            texts.append(node.text)
        if node.tail:
            texts.append(node.tail)
    parts.append("".join(texts))
text = "\n".join(parts)
dst.write_text(text, encoding="utf-8")
print(text)
